### PR TITLE
Add EFA Support for Multinode in AWS

### PR DESCRIPTION
**Pull request by mvpatel2000**
# What does this PR do?

Add EFA Support for Multinode in AWS by updating drivers in docker image. Extends docker matrix to add an EFA/AWS image and updates readme to include new image. 

# What issue(s) does this change relate to?

[CO-1656](https://mosaicml.atlassian.net/browse/CO-1656)

[CO-1656]: https://mosaicml.atlassian.net/browse/CO-1656?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -128,7 +128,32 @@ RUN apt-get update && \
         liblcms2-dev \
         tcl \
         libjpeg8-dev \
-        less && \
+        less \
+        # For AWS EFA:
```
Comment (bandish-shah) [inline docker/Dockerfile:132]: Does it make sense to only install the EFA dependencies on the CUDA images?

**Review by bandish-shah — COMMENTED** [review]
Overall looks good, would be good if you can add some information about what and how this was tested on an AWS EFA deployment.  Going to hold off on the approval until the build passes and the relevant testing info has been added.

**Comment by dakinggg** [comment]
Given how much stuff is getting added, could you compare image size before and after? I wonder if it is enough that we should break the EFA images off

**Comment by mvpatel2000** [comment]
This is a great pr.

**Review by bandish-shah — APPROVED** [review]
LGTM

**Pull request merged**